File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/1. Binary Tree Preorder Traversal/1. Recursive Method/BinaryTreePreorderTraversalRecursive.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    
    // create an array list to hold output values
    List<Integer> output = new ArrayList<>();
    
    public List<Integer> preorderTraversal(TreeNode root) {
        
        // base case: if the root node is null, return the output
        if (root == null) {

            return output;
        }
        
        // add the current node's value to the output
        output.add(root.val);

        // traverse the left node
        preorderTraversal(root.left);

        // traverse the right node
        preorderTraversal(root.right);
        
        // return the output
        return output;
    }
}

File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/1. Binary Tree Preorder Traversal/1. Recursive Method/BinaryTreePreorderTraversalRecursive.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:

        # intialize a list to keep values to return
        output = []

        def pot(node):
            # base case
            if node == None:

                return

            # root
            output.append(node.val)

            # left
            pot(node.left)

            # right
            pot(node.right)

        pot(root)

        return output
